tests/promises/test_concurrent_future.py

```
@@ -0,0 +1,423 @@
+        - The concurrent future correctly reflects Promise state at different points
+    """
+
+    coro_call_count = 0
+
+    # Create a Promise
+    if start_soon is None:
+        # `start_soon=None` in our test means that we want to create a prefilled promise
+        promise = Promise(prefill_result="Hello from Promise!")
+    else:
+
+        async def sample_coro():
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `sample_coro`

Add return type annotation: `str`

(ANN202)